pkg/pool-cow/test/foundry/CowRouter.t.sol

```diff
@@ -0,0 +1,707 @@
+            uint256[] memory donationAfterFees
+        ) = _getDonationAndFees(donationDai, donationUsdc, protocolFeePercentage);
+
+        BaseVaultTest.Balances memory balancesBefore = getBalances(address(cowRouter));
+
+        vm.expectEmit();
+        emit ICowRouter.CoWSwapAndDonation(
+            pool,
+            dai,
+            usdc,
+            usdcSwapAmountOut, // PoolMock is linear, so amounts in == amounts out
+            usdcSwapAmountOut,
+            donationAfterFees,
+            expectedProtocolFees,
+            bytes("")
+        );
+
+        vm.prank(lp);
+        cowRouter.swapExactOutAndDonateSurplus(
+            pool,
+            dai,
+            usdc,
+            MAX_UINT128,
+            usdcSwapAmountOut,
+            type(uint32).max,
+            donationAmounts,
+            bytes("")
+        );
+
+        BaseVaultTest.Balances memory balancesAfter = getBalances(address(cowRouter));
+
+        // Since the pool is linear, amount in == amount out
+        _checkBalancesAfterSwapAndDonation(
+            balancesBefore,
+            balancesAfter,
+            expectedProtocolFees,
+            donationAmounts,
+            usdcSwapAmountOut,
+            usdcSwapAmountOut
+        );
+    }
+
+    /********************************************************
+                            donate()
+    ********************************************************/
+    function testDonate__Fuzz(uint256 donationDai, uint256 donationUsdc, uint256 protocolFeePercentage) public {
+        // ProtocolFeePercentage between 0 and MAX PROTOCOL FEE PERCENTAGE.
+        protocolFeePercentage = bound(protocolFeePercentage, 0, _MAX_PROTOCOL_FEE_PERCENTAGE);
+        donationDai = _boundDonation(donationDai, protocolFeePercentage);
+        donationUsdc = _boundDonation(donationUsdc, protocolFeePercentage);
+
+        vm.prank(admin);
+        cowRouter.setProtocolFeePercentage(protocolFeePercentage);
+
+        (
+            uint256[] memory donationAmounts,
+            uint256[] memory expectedProtocolFees,
+            uint256[] memory donationAfterFees
+        ) = _getDonationAndFees(donationDai, donationUsdc, protocolFeePercentage);
+
+        BaseVaultTest.Balances memory balancesBefore = getBalances(address(cowRouter));
+
+        vm.expectEmit();
+        emit ICowRouter.CoWDonation(pool, donationAfterFees, expectedProtocolFees, bytes(""));
+
+        vm.prank(lp);
+        cowRouter.donate(pool, donationAmounts, bytes(""));
+
+        BaseVaultTest.Balances memory balancesAfter = getBalances(address(cowRouter));
+
+        _checkBalancesAfterSwapAndDonation(balancesBefore, balancesAfter, expectedProtocolFees, donationAmounts, 0, 0);
+    }
+
+    /********************************************************
+                     getProtocolFeePercentage()
+    ********************************************************/
+    function testGetProtocolFeePercentage() public {
```

Comment:
```suggestion
    /********************************************************
                               ProtocolFeePercentage
    ********************************************************/

    function testGetProtocolFeePercentage() public {
```
Block headers are just for a group of functions on a common topic; don't need them for every function.